types/react-native-telnet-client/index.d.ts

```diff
@@ -0,0 +1,106 @@
+// Type definitions for react-native-telnet-client 1.0
+// Project: https://github.com/itgou/react-native-telnet-client
+// Definitions by: Robin Vanden Berghe <https://github.com/robinvandenb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { EventEmitter } from 'events';
+import { Socket, SocketConnectOpts } from 'net';
+import { Duplex, DuplexOptions } from 'stream';
+
+export type Callback<T> = (err: any, value?: T) => void;
+
+export class Stream extends Duplex {
+    constructor(source: Socket, options?: DuplexOptions);
+    _write(data: Buffer | string, encoding?: BufferEncoding, callback?: Callback<void>): void;
+    _read(): void;
+}
+
+export type EscapeHandler = (escapeSequence: string) => string | null;
+
+export type TelnetState =
+    | null
+    | 'end'
+    | 'failedlogin'
+    | 'getprompt'
+    | 'login'
+    | 'ready'
+    | 'response'
+    | 'standby'
+    | 'start';
+
+export interface ExecOptions {
+    echoLines?: number;
+    execTimeout?: number;
+    failedLoginMatch?: string;
+    irs?: string;
+    loginPrompt?: string;
+    maxBufferLength?: number;
+    newlineReplace?: string;
+    ors?: string;
+    shellPrompt?: string;
+    stripControls?: boolean;
+    timeout?: number;
```

Comment:
For this options bag and many more, it's likely the case that you want `| undefined`. See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)